src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb

```diff
@@ -569,7 +569,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                 End If
 
                                 Dim stepOperand As IOperation = If(stepValue.IsConstant, DirectCast(stepValue, IOperation), New Temporary(SyntheticLocalKind.ForLoopStepValue, BoundFor, stepValue))
-                                statements.Add(New CompoundAssignment(controlReference, stepOperand, Semantics.Expression.DeriveAdditionKind(controlType), Nothing, stepValue.Syntax))
+                                statements.Add(New ExpressionStatement(controlReference, stepOperand, Semantics.Expression.DeriveAdditionKind(controlType), Nothing, stepValue.Syntax))
```

Comment:
CompoundAssignment and Assignment was implementing IExpressionStatement. basically it was duplicated code. so I made ExpressionStatement which 2 different constructor. and used it in both places. except the contructor, both were same code. basically exact same code with different field names which implements same IExpressionStatement interface. basically duplicated code.